apis/v1alpha2/httproute_types.go

```diff
@@ -550,7 +572,7 @@ type HTTPRequestHeaderFilter struct {
 	// Support: Extended
 	//
 	// +optional
-	Set map[string]string `json:"set,omitempty"`
+	Set []HTTPHeader `json:"set,omitempty"`
 
 	// Add adds the given header (name, value) to the request
```

Comment:
Tiny nit: I think the pluralization here and for `Add` should match what we have for `Remove`.
```suggestion
	// Add adds the given header(s) (name, value) to the request
```